python/tank/bootstrap/manager.py

```diff
@@ -354,13 +354,43 @@ def bootstrap_engine(self, engine_name, entity=None):
         self._log_startup_message(engine_name, entity)
 
         tk = self._bootstrap_sgtk(engine_name, entity)
-
         engine = self._start_engine(tk, engine_name, entity)
 
         self._report_progress(self.progress_callback, self._BOOTSTRAP_COMPLETED, "Engine launched.")
 
         return engine
 
+    def bootstrap_legacy_shotgun_engine(self, entity=None):
```

Comment:
Let's get rid of the `None` default here. It doesn't make sense to me why we should default to the site config in particular. All values are equivalent, I think there is no concept of a default value for this parameter.